### PR TITLE
draft for error handling module

### PR DESCRIPTION
**Pull request by bbonf**
Trying to come up with a nice way for error reporting.

The current approach here is to show a simple dialog box when running an experiment locally, and to redirect to our usual error page when running online. The redirection URL would include an extra message parameter that can be used to display a relevant message (see https://github.com/UiL-OTS-labs-backoffice/web-experiment-index/pull/3)

### CODE REVIEW

File: jspsych-uil-error.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                      "we'll overwrite it.");
+    }
+
+    uil.error = {};
+
+    (function (context) {
+        context.redirect = function(message) {
+            let encoded = btoa(message);
+            window.location.href = 'https://web-experiments.lab.hum.uu.nl/index_files/error/?msg=' + encoded;
+        };
```
Comment (maartenuni) [inline jspsych-uil-error.js:19]: Perhaps we should keep this method private then, I don't yet mind either

File: jspsych-uil-error.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+if (typeof uil === 'undefined') {
+    console.error("UiL Main library not loaded! " +
+        "Refusing to load uil.error");
+}
+else {
+    if ('error' in uil) {
+        console.warng("uil.error already exists, this is unexpected, " +
```
Comment (maartenuni) [inline jspsych-uil-error.js:9]: Typo

File: jspsych-uil-error.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                }
+                handler(message);
+            });
+            window.addEventListener('unhandledrejection', function (e) {
+                handler(e.reason.message);
+            });
+        };
+    })(uil.error);
+
+    uil.error.registerHandler();
```
Comment (tymees) [inline jspsych-uil-error.js:83]: I'm not entirely sure why'd you make this function even a public namespaced function if you're just going to call it immedialty upon loading?

Comment (bbonf) [reply jspsych-uil-error.js:83]: tbh I wasn't sure if we should have it on by default, or specified in the templates.

Comment (tymees) [reply jspsych-uil-error.js:83]: I think that would be a question for @maartenuni as he's actually writing boilerplates (and I'm not ;) )

Comment (maartenuni) [reply jspsych-uil-error.js:83]: I think i like to have it on. It doesn't hurt and I might provide us with more useful messages as opposed to: Would you please help me, the script suddenly stops. Using uil-utils means using our error handling utilities either, one less condition to worry about.

Comment (maartenuni) [reply jspsych-uil-error.js:83]: Perhaps in the 1.0 release/new module system we could make it more conditionally by (don't) importing the uil.error module.

**Pull request merged**